src/main/java/seedu/address/model/lab/LabList.java

```diff
@@ -187,6 +187,49 @@ public boolean isEmpty() {
         return internalList.isEmpty();
     }
 
+    /**
+     * Aligns the current LabList to the given {@code toAlignWith}.
+     * Aligning LabList means that the current LabList has to have all the Labs that the {@code toAlignWith} has,
+     * and no additional Labs that the {@code toAlignWith} doesn't have.
+     * The end result is that both LabList will have the same list of Labs with potentially different status and marks.
+     * Most relevant for deserialization of labs in {@code JsonSerializableAddressBook}.
+     *
+     * @param toAlignWith The LabList that you want to align to.
+     * @throws DuplicateLabException if one of the LabList contains duplicate labs. (should not be the case)
+     */
+    public void alignLabs(LabList toAlignWith) throws DuplicateLabException {
```

Comment:
From the usages I've seen it seems that `toAlignWith` is always the `MasterLabList`! If this is the case, would it be easier to understand if we rename `toAlignWith` to `masterLabList` or something?

Reply:
Yep it is only used to align with the `MasterLabList`, but I think this functionality is general enough to be placed with `LabList` for extensibility reasons.